src/components/shared/form_fields/DirectDebitField.vue

```diff
@@ -0,0 +1,132 @@
+<template>
+	<div class="form-field-direct-debit">
+		<TextValueField
+			:value="formattedBankNumber"
+			:cursor-position="bankNumberCursorPosition"
+			input-id="account-number"
+			name="account-number"
+			:label="$t( accountNumberLabel )"
+			:placeholder="$t( 'donation_form_payment_bankdata_account_iban_placeholder' )"
+			:error-message="$t( 'donation_form_payment_bankdata_error' )"
+			:show-error="showAccountNumberError"
+			:help-text="$t( 'donation_form_payment_bankdata_account_iban_help_text' )"
+			@field-changed="$emit( 'field-changed', 'account-number' )"
+			@input="onAccountNumberInput"
+		>
+			<template #message>
+				<span v-if="bankName !== '' && accountNumberType === AccountNumberType.IBAN" class="iban-bank-name">{{ bankName }} ({{ bic }})</span>
+			</template>
+		</TextValueField>
+
+		<TextField
+			v-if="accountNumberType === AccountNumberType.Account"
+			v-model="bankCode"
+			input-id="bank-code"
+			name="bank-code"
+			:label="$t( 'donation_form_payment_bankdata_bank_legacy_label' )"
+			:placeholder="$t( 'donation_form_payment_bankdata_bank_legacy_placeholder' )"
+			error-message=""
+			:show-error="showBankCodeError"
+			:help-text="$t( 'donation_form_payment_bankdata_bank_legacy_help_text' )"
+			@field-changed="$emit( 'field-changed', 'bank-code' )"
+			@update:model-value="onBankCodeUpdate"
+		>
+			<template #message>
+				<span v-if="bankName !== ''" class="bank-name">{{ bankName }}</span>
```

Comment:
When pasting an Austrian IBAN the server result comes back without the `bankName` field, making `bankName` `undefined` and this will be rendered none the less. I suggest both a more lenient way of checking (`v-if="bankName"` will accept any falsy value including `undefined`) and a default value in the resource.